### DISCUSSION TITLE
Deep linking routes

### DISCUSSION THREAD

**Discussion by lukeredpath** [pointfreeco/episode-code-samples#105] [General]
In episode 9, I'm curious why you chose to introduce completely separate `Route` enums e.g. `InventoryRoute` rather than re-using the existing route types that were defined in each feature, e.g. `InventoryViewModel.Route`? Do you believe there to be a particular benefit in this, given the additional overhead of maintaining two very similar types?

**Comment by mbrandonw** [comment]
It may be possible to do that, and it would definitely be great to unify the two enums into a single enum. In practice we have found that the URL routes can slightly deviate from the navigation structure (either from the beginning, or over time they drift apart), which makes decoupling them a little more future-proof. If explore unifying the route and navigation enums in our demo application we'd definitely be interested in hearing about the pros and cons you found.
